src/rebar3_hank_prv.erl

```diff
@@ -120,6 +127,87 @@ format_result(#{file := File,
                 text := Msg}) ->
     hank_utils:format_text("~ts:~tp: ~ts", [File, Line, Msg]).
 
+-spec write_data_to_json_file([hank_rule:result()], rebar_state:t()) -> ok.
+write_data_to_json_file(Result, State) ->
```

Comment:
This seems like a conditional operation. Maybe naming the function `maybe_write_data_to_json_file` would be more explicit?